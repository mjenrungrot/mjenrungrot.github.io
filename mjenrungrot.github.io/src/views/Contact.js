// @flow

// eslint-disable-next-line no-unused-vars
import React from 'react';
// eslint-disable-next-line no-unused-vars
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';

import {withStyles} from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import Fade from '@material-ui/core/Fade';
import Typography from '@material-ui/core/Typography';
import Divider from '@material-ui/core/Divider';

import Main from '../layouts/Main';

const styles = (theme) => ({
  root: {
    ...theme.mixins.gutters(),
    paddingTop: theme.spacing(2),
    paddingBottom: theme.spacing(2),
  },
  address: {
    marginLeft: theme.spacing(3),
  },
});

/**
 * Contact View Component
 * @param {object} props creation params
 * @return {object} Index component
 */
function Contact(props) {
  const {classes} = props;
  return (
    <Main selectedTab="/contact" infoBar showContact>
      <Fade in>
        <Paper className={classes.root}>
          <Typography variant="h5" gutterBottom>
            Contact
          </Typography>
          <Divider />
          <Typography variant="body1" paragraph>
            There are multiple ways to contact me.
          </Typography>
          <Typography paragraph>
            If you&apos;re reaching out for work opportunities, please
            reach out via LinkedIn message or my school email and
            simply indicate that you follow from my website.
          </Typography>
          <Typography paragraph>
            If you&apos;re interested in research collaboration, please
            reach out via school email. I&apos;m very open to research
            in many areas related to multimedia, AR/VR, deep learning,
            music information retrieval, computer vision, computational
            photography, and more.
          </Typography>
          <Typography paragraph>
            Lastly, if you need to mail or send relevant package,
            please refer to the following address.
          </Typography>
          <br />
          <Typography className={classes.address}>
            Teerapat Jenrungrot<br />
            Bill and Melinda Gates Center for Computer Science and
            Engineering<br />
            3800 E Stevens Way NE<br />
            Seattle, WA 98195
          </Typography>
        </Paper>
      </Fade>
    </Main>
  );
}

Contact.propTypes = {
  classes: PropTypes.any.isRequired,
};

export default withStyles(styles)(Contact);
