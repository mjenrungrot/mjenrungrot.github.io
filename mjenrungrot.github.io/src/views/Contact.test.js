// eslint-disable-next-line no-unused-vars
import React from 'react';
import {shallow} from 'enzyme';
import Contact from './Contact';

describe('Contact component', () => {
  test('renders', () => {
    const component = shallow(<Contact />);
    expect(component).toMatchSnapshot();
  });
});
