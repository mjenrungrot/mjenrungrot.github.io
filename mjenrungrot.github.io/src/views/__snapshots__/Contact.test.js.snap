// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact component renders 1`] = `
<Contact
  classes={
    Object {
      "address": "Contact-address-2",
      "root": "Contact-root-1",
    }
  }
/>
`;
